src/app/api/core/utils/withRetry.ts

```diff
@@ -0,0 +1,49 @@
+import { StatusableError } from '@/types/CopilotApiError'
+import pRetry, { FailedAttemptError } from 'p-retry'
+import * as Sentry from '@sentry/nextjs'
+
+export const withRetry = async <T>(fn: (...args: any[]) => Promise<T>, args: any[]): Promise<T> => {
```

Comment:
The `fn` parameter can take any number of arguments due to the `(...args: any[])` definition. This is fine but I think it does reduce type safety.
If possible, let's define the function signature more explicitly based on your use case for better type safety.

Reply:
Using a generic makes it unusable for our needs. Do you have any suggestion here?